packages/teuchos/parameterlist/test/ParameterList/ParameterList_UnitTests.cpp

```diff
@@ -442,6 +442,47 @@ TEUCHOS_UNIT_TEST( ParameterList, set_get_string )
 
 }
 
+TEUCHOS_UNIT_TEST( ParameterList, set_string_move_semantics)
+{
+  ParameterList pl;
+
+  ECHO(std::string my_str_1{"my text 1"});
+  ECHO(pl.set("my string 1", std::move(my_str_1)));
+
+  // Check that the parameter value was moved by checking that my_str_1 is now empty.
+  TEST_ASSERT(my_str_1.empty());
+
+  TEST_EQUALITY_CONST(pl.get<std::string>("my string 1"), "my text 1");
+}
+
+TEUCHOS_UNIT_TEST( ParameterList, set_string_specified_template_argument)
+{
+  // Check the templated set method and its overload when the template argument is specified.
+
+   ParameterList pl;
+
+  // The parameter value can be passed by rvalue reference.
+  // The main templated set method is called, and the parameter value is moved.
+  ECHO(std::string my_str_2{"my text 2"});
+  ECHO(pl.set<std::string>("my string 2", std::move(my_str_2)));
+  TEST_ASSERT(my_str_2.empty());
+  TEST_EQUALITY_CONST(pl.get<std::string>("my string 2"), "my text 2");
+  
+  // The parameter value cannot be passed by rvalue reference.
+  // The overload of the templated set method is called, and the parameter value is not moved.
+  ECHO(std::string my_str_3{"my text 3"});
+  ECHO(pl.set<std::string>("my string 3", my_str_3));
+  TEST_ASSERT( ! my_str_3.empty());
+  TEST_EQUALITY_CONST(pl.get<std::string>("my string 3"), "my text 3");
+
+  ECHO(const std::string my_str_4{"my text 4"});
+  ECHO(pl.set<std::string>("my string 4", my_str_4));
```

Comment:
So `my_str_4` is actually passed as `const std::string&`?

Reply:
I agree. We can expect it to be passed by const ref.